### PR TITLE
Actions Module: Withdraw Args

### PR DESCRIPTION
**Pull request by aparnakr**
This PR adds the following:

WithdrawArgs struct
function to parse the withdraw arguments

### CODE REVIEW

File: contracts/libs/Actions.sol [hunk 1]
```diff
@@ -84,4 +102,27 @@ library Actions {
...
+     * @notice Parses the passed in action argmuents to get the argmuents for a withdraw action
+     * @param _args The general action arguments structure
+     * @return The arguments for a withdraw action
+     */
+    function _parseWithdrawArgs(ActionArgs memory _args) internal returns (WithdrawArgs memory) {
+        require(
+            (_args.actionType == ActionType.WithdrawLongOption) || (_args.actionType == ActionType.WithdrawCollateral),
+            "Actions: can only parse arguments for withdraw actions"
+        );
+        require(_args.owner != address(0), "Actions: cannot withdraw to an invalid account");
```
Comment (antoncoding) [inline contracts/libs/Actions.sol:116]:
```suggestion
        require(_args.owner != address(0), "Actions: cannot withdraw from an invalid vault");
        require(_args.sender != address(0), "Actions: cannot withdraw to an invalid account");
```

File: test/Actions.test.ts [hunk 2]
```diff
@@ -130,4 +130,109 @@ contract('Actions', ([owner, random]) => {
...
+        owner: ZERO_ADDR,
+        sender: owner,
+        asset: asset,
+        vaultId: vaultId,
+        amount: amount,
+        index: index,
+        data: bytesArgs,
+      }
+
+      await expectRevert(actionTester.testParseWithdrawAction(data), 'Actions: cannot withdraw to an invalid account')
```
Comment (antoncoding) [inline test/Actions.test.ts:177]: This is actually testing an invalid `owner` address. 
Should have both `args.owner` and `args.sender` checks in the contract (As suggested above) and have 2 separate tests here.

Comment (aparnakr) [reply test/Actions.test.ts:177]: hmm what do you mean?

Comment (aparnakr) [reply test/Actions.test.ts:177]: Should the check be happening here or in the controller?

Comment (antoncoding) [reply test/Actions.test.ts:177]: You have 2 require checks right?
require(_args.owner != address(0), "Actions: cannot withdraw from an invalid vault");
require(_args.sender != address(0), "Actions: cannot withdraw to an invalid account");

Just saying you need test for both of them

**Comment by haythem96** [comment]
Will be great if we fix conflicts here and merge this one.

File: contracts/libs/Actions.sol [hunk 3]
```diff
@@ -62,6 +62,24 @@ library Actions {
         uint256 amount;
     }
 
+    struct WithdrawArgs {
+        /// @notice The address of the account owner
```
Comment (antoncoding) [inline contracts/libs/Actions.sol:66]:
```suggestion
        // The address of the account owner
```

File: contracts/libs/Actions.sol [hunk 4]
```diff
@@ -84,4 +102,28 @@ library Actions {
                 amount: _args.amount
             });
     }
+
+    /**
+     * @notice Parses the passed in action argmuents to get the argmuents for a withdraw action
+     * @param _args The general action arguments structure
+     * @return The arguments for a withdraw action
+     */
+    function _parseWithdrawArgs(ActionArgs memory _args) internal returns (WithdrawArgs memory) {
```
Comment (antoncoding) [inline contracts/libs/Actions.sol:111]:
```suggestion
    function _parseWithdrawArgs(ActionArgs memory _args) internal pure returns (WithdrawArgs memory) {
```

**Pull request merged**